lib/spack/docs/packaging_guide.rst

```diff
@@ -679,7 +679,14 @@ http://example.com/foo-8.2.1.tar.gz.
 
 If the URL is particularly complicated or changes based on the release,
 you can override the default URL generation algorithm by defining your
-own ``url_for_version()`` function. For example, the download URL for
+own ``url_for_version()`` function. 
+
+.. warning:
+
+   ``spack versions`` and ``spack checksum`` do not use the custom ``url_for_version()``.
```

Comment:
> do not use

this isn't correct. Maybe just emphasize the difference between

```
Hm, this is not entirely right either. Maybe just emphasize the difference between

```
spack checksum python@3.12  # it crawls/spiders web pages for links to new versions
spack checksum python 3.12.0 # it fetches a single version, the URL extrapolated from `url` / `url_for_version`.
```